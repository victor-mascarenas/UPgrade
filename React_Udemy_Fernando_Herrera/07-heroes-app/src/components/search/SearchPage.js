import React from 'react';
import useForm from '../../hooks/useForm';
import { getHeroesByName } from '../../selectors/getHeroesByName';
import { HeroCard } from '../hero/HeroCard';

export const SearchPage = () => {
    const [{searchText}, inputOnChange] = useForm({
        searchText: ''
    });

    const heroesFiltered = getHeroesByName('algo');

    const handleSearch = (e) => {
        e.preventDefault();
        console.log(searchText);
    };

    return (
        <>
            <h1>Busqueda</h1>
            <hr/>
            <div className='row'>
                <div className='col-5'>
                    <h4>Buscar</h4>
                    <hr/>
                    <form onSubmit={handleSearch}>
                        <input type='text' placeholder='Buscar un heroe' className="form-control" name='searchText' autoComplete='off' value={searchText} onChange={inputOnChange}/>
                        <button type='submit' className='btn btn-outline-primary mt-3 col-12'>Buscar</button>
                    </form>
                </div>
                <div className='col-7'>
                    <h4>Resultados</h4>
                    <hr/>
                    {
                        heroesFiltered.map(hero => {
                            return <HeroCard key={hero.id} hero={hero}/>;
                        })
                    }
                </div>
            </div>
        </>
    );
};
