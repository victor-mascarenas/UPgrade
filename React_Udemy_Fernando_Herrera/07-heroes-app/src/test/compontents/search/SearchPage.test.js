import { SearchPage } from "../../../components/search/SearchPage";
import { mount } from 'Enzyme';
import { MemoryRouter } from "react-router-dom";

describe('Pruebas en SearchPage', () => {
    test('Debe de mostrarse correctamente con valores por defecto', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar mensaje "sin resultados"', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>
        );
        expect(wrapper.find('.card-text').text().trim()).toBe('Ingresa un termino de busqueda');
    });
});