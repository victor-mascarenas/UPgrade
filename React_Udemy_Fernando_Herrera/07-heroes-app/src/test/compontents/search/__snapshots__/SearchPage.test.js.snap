// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchPage Debe de mostrarse correctamente con valores por defecto 1`] = `
Array [
  <h1>
    Busqueda
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Buscar un heroe"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-3 col-12"
          type="submit"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="mt-3 card bg-warning text-dark mb-3"
      >
        <div
          className="card-header"
        >
          Resultados de busqueda
        </div>
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Sin resultados
          </h5>
          <p
            className="card-text"
          >
            Ingresa un termino de busqueda
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
